src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import React, { useCallback, useEffect, useState } from 'react';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { TodosStatus } from './types/TodosStatus';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodosStatus, setSelectedTodosStatus] = useState<TodosStatus>(
+    TodosStatus.All,
+  );
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setIsTodosLoaded(true);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+  }, []);
+
+  const chooseUser = useCallback((todo: Todo | null) => {
+    setSelectedTodo(todo);
+  }, []);
+
+  const filterTodos = () => {
```

Comment:
```suggestion
  const filterTodos = (todos: Todo[], status: TodosStatus) => {
```
I see you use `useCallback`, so a few words about optimization (just FYI):
- let the function accept 2 arguments :arrow_up: 
- move it out of component
- create in component `const visibleTodos = useMemo(() => filterTodos(todos, selectedTodosStatus), [todos, selectedTodosStatus])`

=> visible todos will be recalculated only on `todos` or  `selectedTodosStatus` change